### PR TITLE
simplify amountOff calc

### PR DESCRIPTION
**Pull request by mattkrick**
simplified the logic a bit @ackernaut for your consideration

### CODE REVIEW

File: packages/client/modules/invoice/components/Invoice/Invoice.tsx [hunk 1]
```diff
@@ -174,10 +173,11 @@ const Invoice = (props: Props) => {
     tier
   } = invoiceDetails
   const status = invoiceDetails.status as InvoiceStatusEnum
-  const {interval, nextPeriodEnd} = nextPeriodCharges!
+  const {amount, interval, nextPeriodEnd} = nextPeriodCharges!
   const chargeDates = `${makeDateString(startAt)} to ${makeDateString(endAt)}`
   const nextChargesDates = `${makeDateString(endAt)} to ${makeDateString(nextPeriodEnd)}`
-  const discountedAmount = coupon ? `-${coupon.amountOff && invoiceLineFormat(coupon.amountOff) || coupon.percentOff && invoicePercentToDiscountedAmount(nextPeriodCharges.amount, coupon.percentOff)}` : ''
+  const amountOff = coupon && (coupon.amountOff || coupon.percentOff! / 100 * amount)
```
Comment (ackernaut) [inline packages/client/modules/invoice/components/Invoice/Invoice.tsx]: @mattkrick beautiful—what’s the bang `!` after percentOff do?

Comment (ackernaut) [reply packages/client/modules/invoice/components/Invoice/Invoice.tsx]: I guess this: https://stackoverflow.com/questions/42273853/in-typescript-what-is-the-exclamation-mark-bang-operator-when-dereferenci

Comment (ackernaut) [reply packages/client/modules/invoice/components/Invoice/Invoice.tsx]: TIL: non-null assertion operator

Comment (mattkrick) [reply packages/client/modules/invoice/components/Invoice/Invoice.tsx]: yep! typescript doesn't offer mutually exclusive conditions, but _we_ know that if percentOff doesn't exist, amountOff does & vice versa, so it's safe to assume it's non null

**Pull request merged**